lib/semantics/tools.cc

```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "tools.h"
 #include "scope.h"
+#include "tools.h"
```

Comment:
"tools.h" is supposed to be the first include and clang-format is supposed to enforce that.
To do that it has to know that this file is "tools.cc", so "clang-format < tools.cc" gives a different result than "clang-format -i tools.cc".

Reply:
Today I learned about git-clang-format.

Reply:
This is what I use in vim:
```
nmap ,cf :update<cr>:silent !clang-format -i %<cr>:e %<cr>
```
Then `,cf` reformats the current file.